tests/ignite/contrib/handlers/test_lr_finder.py

```diff
@@ -462,4 +463,130 @@ def test_plot_multiple_param_groups(
+    assert os.path.exists("dummy_muliple_param_groups2.jpg")
+
+
+def _test_distrib_log_lr_and_loss(device):
+    from ignite.handlers import PiecewiseLinear
+
+    class DummyModel(nn.Module):
+        def __init__(self, n_channels=10, out_channels=10, flatten_input=False):
+            super(DummyModel, self).__init__()
+
+            self.net = nn.Sequential(
+                nn.Flatten() if flatten_input else nn.Identity(), nn.Linear(n_channels, out_channels)
+            )
+
+        def forward(self, x):
+            return self.net(x)
+
+    model = DummyModel()
+    optimizer = SGD(model.parameters(), lr=1e-4, momentum=0.0)
+    lr_finder = FastaiLRFinder()
+
+    # minimal setup for lr_finder to make _log_lr_and_loss work
+    rank = idist.get_rank()
+    loss = 0.01 * (rank + 1)
+
+    _lr_schedule = PiecewiseLinear(optimizer, param_name="lr", milestones_values=[(0, 1e-4), (1, 10)])
+
+    engine = Engine(lambda e, b: None)
+
+    engine.state.output = loss
+    engine.state.iteration = 1
+    lr_finder._lr_schedule = _lr_schedule
+    lr_finder._history["loss"] = []
+    lr_finder._history["lr"] = []
+
+    lr_finder._log_lr_and_loss(engine, output_transform=lambda x: x, smooth_f=0.1, diverge_th=10.0)
+    # expected_loss = loss
+    # for i in range(rank - 1):
+    #     expected_loss += 0.01 * (i + 1)
+    expected_loss = idist.all_reduce(loss)
+    assert pytest.approx(lr_finder._history["loss"][-1]) == expected_loss
+
+
+def _test_distrib_integration_mnist(device):
+    from torch.utils.data import DataLoader
+    from torchvision.datasets import MNIST
+    from torchvision.transforms import Compose, Normalize, ToTensor
+
+    data_transform = Compose([ToTensor(), Normalize((0.1307,), (0.3081,))])
+
+    train_loader = DataLoader(
+        MNIST(download=True, root="/tmp", transform=data_transform, train=True), batch_size=256, shuffle=True
+    )
+
+    class DummyModel(nn.Module):
+        def __init__(self, n_channels=10, out_channels=1, flatten_input=False):
+            super(DummyModel, self).__init__()
+
+            self.net = nn.Sequential(
+                nn.Flatten() if flatten_input else nn.Identity(), nn.Linear(n_channels, out_channels)
+            )
+
+        def forward(self, x):
+            return self.net(x)
+
+    model = DummyModel(n_channels=784, out_channels=10, flatten_input=True)
+    model = model.to(device)
+
+    optimizer = SGD(model.parameters(), lr=1e-4, momentum=0.0)
+    to_save = {"model": model, "optimizer": optimizer}
+    engine = create_supervised_trainer(model, optimizer, nn.CrossEntropyLoss())
+    lr_finder = FastaiLRFinder()
+    with lr_finder.attach(engine, to_save) as trainer_with_finder:
+        trainer_with_finder.run(train_loader)
+
+    lr_finder.plot()
+    ax = lr_finder.plot(skip_end=0)
+    ax.figure.savefig("distrib_dummy.jpg")
+    assert os.path.exists("distrib_dummy.jpg")
+
+    assert 1e-4 <= lr_finder.lr_suggestion() <= 2
```

Comment:
Let's reduce the interval of expected values and let's add `lr_finder.apply_suggested_lr` to check